server/system/service/smtp_configuration_checker.go

```diff
@@ -0,0 +1,215 @@
+	var (
+		tlsConfig = &tls.Config{}
+	)
+
+	checkResults = &types.SmtpCheckResult{}
+
+	if smtpConfigs.Port == 0 {
+		smtpConfigs.Port = 25
+	}
+
+	//check for validity of the host
+	if !svc.isValidHost(smtpConfigs.Host) {
+		checkResults.Host.Message = fmt.Sprintf("%s name is Invalid", smtpConfigs.Host)
+	}
+
+	if checkResults.Host.Message == "" {
+		checkResults.Host.Success = true
+	}
+
+	// Applying TLS
+	tlsConfig = &tls.Config{ServerName: smtpConfigs.Host}
+
+	if smtpConfigs.TLSInsecure {
+		tlsConfig.InsecureSkipVerify = true
+	}
+
+	if smtpConfigs.TLSServerName != "" {
+		tlsConfig.ServerName = smtpConfigs.TLSServerName
+	}
+
+	checkResults.Server.Message = mail.SMTPConfigurationCheck(smtpConfigs.Host, smtpConfigs.Port, smtpConfigs.Username, smtpConfigs.Password, tlsConfig)
+	if checkResults.Server.Message == "" {
+		checkResults.Server.Success = true
+	}
+
+	//send the email there are recipients
+	if len(smtpConfigs.Recipients) != 0 {
+		checkResults.Send.Message, err = svc.smtpSend(ctx, smtpConfigs.Recipients)
+		if checkResults.Send.Message == "" && err == nil {
+			checkResults.Send.Success = true
+		}
+	}
+
+	return checkResults, err
+}
+
+func (svc smtpConfigurationChecker) smtpSend(ctx context.Context, recipients []string) (expected string, err error) {
+	var (
+		ntf      = mail.New()
+		toHeader string
+		// context with service user
+		// we need this for retrieving & rendering email templates
+		suCtx = intAuth.SetIdentityToContext(ctx, intAuth.ServiceUser())
+	)
+
+	if err = svc.procEmailRecipients(ntf, "To", recipients); err != nil {
+		return "", err
+	}
+
+	toHeader = strings.Join(recipients, ",")
+	ntf.SetAddressHeader("To", toHeader, "")
+
+	st, ct, err := svc.findEmailTemplates(suCtx)
+	// if we cannot find an email template
+	if err != nil {
+		ntf.SetHeader("Subject", "SMTP Configuration check")
+		ntf.SetBody("text/html", "<h2 style=\"color: #568ba2;text-align: center;\">SMTP configurations check passed</h2>")
+
+        err = mail.Send(ntf)
+
+        if err != nil {
+            return err.Error(), nil
+        }
+
+        return "", nil
+    }
+
+	subjectTmp, contentTmp, err := svc.procEmailTemplate(suCtx, st.ID, ct.ID)
+	if err != nil {
+		return "", err
+	}
+
+	ntf.SetHeader("Subject", string(subjectTmp))
+	ntf.SetBody("text/html", string(contentTmp))
+
+	err = mail.Send(ntf)
+
+	if err != nil {
+		return err.Error(), nil
+	}
+
+	return "", nil
+}
+
+func (svc smtpConfigurationChecker) procEmailRecipients(m *gomail.Message, field string, recipients []string) (err error) {
+	var (
+		email string
+		rcpt  string
+	)
+
+	if len(recipients) == 0 {
+		return
+	}
+
+	for _, rcpt = range recipients {
+
+		email = strings.TrimSpace(rcpt)
+
+		// Validate email here
+		if !mail.IsValidAddress(email) {
+			return fmt.Errorf("invalid recipient email address %s", email)
+		}
+
+	}
+
+	m.SetHeader(field, recipients...)
+	return nil
+}
+
+func (svc smtpConfigurationChecker) isValidHost(host string) bool {
```

Comment:
- move this to pkg/mail as a standalone function, export it
 - add some tests (look at `mail.TestMailValidator` for inspiration)